File: solutions/python/0001_two_sum.py
from typing import List

def two_sum(nums: List[int], target: int) -> List[int]:
    """
    两数之和
    :param nums: 整数数组
    :param target: 目标值
    :return: 两个数的索引
    """
    num_map = {}
    for i, num in enumerate(nums):
        complement = target - num
        if complement in num_map:
            return [num_map[complement], i]
        num_map[num] = i
    return []

if __name__ == "__main__":
    import json
    with open("../../problems/0001-two-sum/test_cases.json", "r") as f:
        test_cases = json.load(f)["test_cases"]
    
    for i, test_case in enumerate(test_cases, 1):
        input_data = test_case["input"]
        expected = test_case["expected"]
        result = two_sum(input_data["nums"], input_data["target"])
        print(f"测试用例 {i}:")
        print(f"输入: nums = {input_data['nums']}, target = {input_data['target']}")
        print(f"输出: {result}")
        print(f"预期: {expected}")
        print(f"结果: {'通过' if result == expected else '失败'}\n") 

File: solutions/rust/src/0001_two_sum.rs
use std::collections::HashMap;

pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
    let mut map = HashMap::new();
    
    for (i, &num) in nums.iter().enumerate() {
        let complement = target - num;
        if let Some(&j) = map.get(&complement) {
            return vec![j as i32, i as i32];
        }
        map.insert(num, i);
    }
    
    vec![]
}

#[cfg(test)]
mod tests {
    use super::*;
    use test_case::test_case;

    #[test_case(vec![2, 7, 11, 15], 9, vec![0, 1])]
    #[test_case(vec![3, 2, 4], 6, vec![1, 2])]
    #[test_case(vec![3, 3], 6, vec![0, 1])]
    fn test_two_sum(nums: Vec<i32>, target: i32, expected: Vec<i32>) {
        assert_eq!(two_sum(nums, target), expected);
    }
} 